aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -92,11 +95,30 @@ synchronized void startSubscriptions() throws DataStoreException {
         buffer = ReplaySubject.create();
 
         Set<Observable<SubscriptionEvent<? extends Model>>> subscriptions = new HashSet<>();
-        for (Class<? extends Model> clazz : modelProvider.models()) {
-            for (SubscriptionType subscriptionType : SubscriptionType.values()) {
-                subscriptions.add(subscriptionObservable(appSync, subscriptionType, latch, clazz));
+        if (modelProvider.models() == null || modelProvider.models().isEmpty()) {
+            for (ModelSchema modelSchema : modelProvider.modelSchemas().values()) {
+                for (SubscriptionType subscriptionType : SubscriptionType.values()) {
+                    subscriptions.add(
+                            subscriptionObservable(
+                                    appSync,
+                                    subscriptionType,
+                                    latch,
+                                    SerializedModel.class,
+                                    modelSchema
+                            )
+                    );
+                }
+            }
+        } else {
```

Comment:
If modelProvider doesn't vend models but only schemas (i.e. Flutter)

Reply:
To avoid the logic fork here, could you instead:
 - Add modelClass as a property on `ModelSchema`.  For Flutter, this would always be SerializedModel.class.
 - Then you can just have:

```
for (ModelSchema modelSchema : modelProvider.modelSchemas().values()) {
                for (SubscriptionType subscriptionType : SubscriptionType.values()) {
                    subscriptions.add(
                            subscriptionObservable(
                                    appSync,
                                    subscriptionType,
                                    latch,
                                    modelSchema.getClass(),
                                    modelSchema
                            )
                    );
                }
            }
```

And actually, then, you would not even need to pass the model Class as a parameter, since you can get it from the ModelSchema internally if needed.

Reply:
That's a very good idea. Updated all the references.